api/schema_responses/serializers.py

```diff
@@ -0,0 +1,121 @@
+from api.base.utils import absolute_reverse
+from api.base.serializers import JSONAPISerializer, LinksField
+from rest_framework import serializers as ser
+from rest_framework import exceptions
+
+from api.base.serializers import (
+    RelationshipField,
+    VersionedDateTimeField,
+)
+
+from osf.models import (
+    Registration,
+    SchemaResponse,
+    RegistrationSchema,
+)
+
+
+class RegistrationSchemaResponseSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'date_created',
+        'date_modified',
+        'revision_justification',
+        'reviews_state',
+    ])
+    writeable_method_fields = frozenset([
+        'revision_response',
+    ])
+
+    id = ser.CharField(source='_id', required=False, allow_null=True)
+    date_created = VersionedDateTimeField(source='created', required=False)
+    date_submitted = VersionedDateTimeField(source='submitted_timestamp', required=False)
+    date_modified = VersionedDateTimeField(source='modified', required=False)
+    revision_justification = ser.CharField(required=False)
+    updated_response_keys = ser.JSONField(required=False, read_only=True)
+    reviews_state = ser.ChoiceField(choices=['revision_in_progress', 'revision_pending_admin_approval', 'revision_pending_moderation', 'approved'], required=False)
+    is_pending_current_user_approval = ser.SerializerMethodField()
+    revision_responses = ser.SerializerMethodField()
```

Comment:
I think we discussed this before, but does this work as a JsonField with `source=all_responses`?

Reply:
It works, but still needs to be listed `writeable_method_fields` as a JSONField for some reason, so I'll change it, but it's still not 100% right.